src/core/commit.rs

```diff
@@ -443,8 +445,29 @@ impl<Witness, App: Application> CommitNode<Witness, App> {
 
 impl<App: Application> CommitNode<(), App> {
     /// Decode a Simplicity program from bits, without the witness data.
+    /// Decode a Simplicity program from bits, without the witness data.
+    ///
+    /// # Usage
+    ///
+    /// Use this method only if the serialization **does not** include the witness data.
+    /// This means, the program simply has no witness during commitment,
+    /// or the witness is provided by other means.
+    ///
+    /// If the serialization contains the witness data, then use [`Node::decode()`].
     pub fn decode<I: Iterator<Item = u8>>(bits: &mut BitIter<I>) -> Result<Rc<Self>, Error> {
-        decode::decode_program_no_witness(bits)
+        decode::decode_program_no_witness(bits, true)
```

Comment:
In 6076bdbfe1e6624aa1c09ed9021950fe1952694c:

This boolean argument shouldn't be exposed outside the `decode` module. It's fine to have one function so you can reuse a bunch of the code but there should be two separate functions exposed from `decode::`.